### ISSUE TITLE
write on object references should produce an error

### ISSUE THREAD

**Issue by arcanist123** [abaplint/abaplint#1983]
This report is recognised as a correct report. It should show an error
``` abap
REPORT z_test_transpiler.

CLASS lcl_ DEFINITION CREATE PUBLIC.

  PUBLIC SECTION.
ENDCLASS.

CLASS lcl_ IMPLEMENTATION.
ENDCLASS.

START-OF-SELECTION.
  DATA lo_aaa TYPE REF TO lcl_.
  WRITE: lo_aaa.
```

**Comment by larshp** [comment]
@arcanist123 try again, this was fixed some time ago,

![image](https://user-images.githubusercontent.com/5888506/163802974-270de8c5-cb9e-428f-ae85-4590625b69c7.png)

**Comment by larshp** [comment]
closing, no reply
